### PR TITLE
ARROW-12406: [R] Fix checkbashism violation in configure

### PR DESCRIPTION
**Pull request by jonkeane**
Fix checkbashim foo=bar; export foo

### CODE REVIEW

**Comment by jonkeane** [comment]
R CI from my branch:
https://github.com/jonkeane/arrow/actions/runs/752347395

And crossbow manual build (there is still noise and a separate failure, but this issue is fixed):
https://github.com/ursacomputing/crossbow/runs/2353713416?check_suite_focus=true

**Comment by jonkeane** [comment]
What remains of the checkbashism has been around since February (before we added the line that this PR fixes)

https://github.com/ursacomputing/crossbow/runs/1985368794?check_suite_focus=true#step:7:181

File: r/configure [hunk 1]
```diff
@@ -148,7 +148,7 @@ else
       if [ "$UNAME" = "Darwin" ] && [ "${OPENSSL_ROOT_DIR}" = "" ]; then
         brew --prefix openssl >/dev/null 2>&1
         if [ $? -eq 0 ]; then
-          export OPENSSL_ROOT_DIR="$(brew --prefix openssl)"
+          OPENSSL_ROOT_DIR="$(brew --prefix openssl)"; export OPENSSL_ROOT_DIR
```
Comment (nealrichardson) [inline r/configure:151]: (insert BDR quote about how `$()` was a completely unnecessary change in bash)

```suggestion
          OPENSSL_ROOT_DIR="`brew --prefix openssl`"; export OPENSSL_ROOT_DIR
```

**Review by pitrou — APPROVED** [review]
+1
